src/Ryujinx.HLE/HOS/ModLoader.cs

```diff
@@ -238,47 +269,85 @@ private static void QueryPatchDirs(PatchCache cache, DirectoryInfo patchDir)
 
             foreach (var modDir in patchDir.EnumerateDirectories())
             {
-                patches.Add(new Mod<DirectoryInfo>(modDir.Name, modDir));
+                patches.Add(new Mod<DirectoryInfo>(modDir.Name, modDir, true));
                 Logger.Info?.Print(LogClass.ModLoader, $"Found {type} patch '{modDir.Name}'");
             }
         }
 
-        private static void QueryTitleDir(ModCache mods, DirectoryInfo titleDir)
+        private static void QueryApplicationDir(ModCache mods, DirectoryInfo applicationDir, ulong applicationId)
         {
-            if (!titleDir.Exists)
+            if (!applicationDir.Exists)
             {
                 return;
             }
 
-            var fsFile = new FileInfo(Path.Combine(titleDir.FullName, RomfsContainer));
+            string modJsonPath = Path.Combine(AppDataManager.GamesDirPath, applicationId.ToString("x16"), "mods.json");
+            ModMetadata modMetadata = new();
+
+            try
+            {
+                modMetadata = JsonHelper.DeserializeFromFile(modJsonPath, _serializerContext.ModMetadata);
+            }
+            catch
+            {
+                Logger.Warning?.Print(LogClass.ModLoader, $"Failed to deserialize mod data for {applicationId} at {modJsonPath}");
```

Comment:
```suggestion
                Logger.Warning?.Print(LogClass.ModLoader, $"Failed to deserialize mod data for {applicationId:X16} at {modJsonPath}");
```
It should print the application ID as hex.

Reply:
I think it also prints this when the file doesn't exist yet, not sure if that is intentional?